test/e2e/branding_test.go

```diff
@@ -0,0 +1,168 @@
+  documentationBaseURL: https://docs.okd.io/4.0/
+`,
+			},
+		}
+		_, err = modifyConsoleConfigMap(client, consoleapi.OpenshiftConfigManagedNamespace, cm)
+		if err != nil {
+			t.Fatalf("error: could not apply managed config map %v", err)
+		}
+
+		err = wait.Poll(1*time.Second, pollTimeout, func() (stop bool, err error) {
+			gotBrand := getConsoleBrand(t, client)
+			return (gotBrand == expectedBrand), nil
+		})
+	}
+}
+
+// Set Brand on the operator config
+func setOperaterConfigBrand(t *testing.T, client *testframework.Clientset, brand operatorsv1.Brand) {
+	operatorConfig, err := client.Consoles().Get(consoleapi.ConfigResourceName, metav1.GetOptions{})
+	if err != nil {
+		t.Fatalf("could not get operator config, %v", err)
+	}
+	// set it
+	spec := operatorsv1.ConsoleSpec{
+		OperatorSpec: operatorsv1.OperatorSpec{
+			ManagementState: "Managed",
+		},
+		Customization: operatorsv1.ConsoleCustomization{
+			Brand:                brand,
+			DocumentationBaseURL: "https://docs.okd.io/4.1/",
+		},
+	}
+	operatorConfig.Spec = spec
+	_, err = client.Consoles().Update(operatorConfig)
+	if err != nil {
+		t.Fatalf("could not update operator config with brand=%v, %v", brand, err)
+	}
+}
+
+// Get Brand on the operator config
+func getOperaterConfigBrand(t *testing.T, client *testframework.Clientset) operatorsv1.Brand {
+	operatorConfig, err := client.Consoles().Get(consoleapi.ConfigResourceName, metav1.GetOptions{})
+	if err != nil {
+		t.Fatalf("Could not get operator config, %v", err)
+	}
+	// set it
+	return operatorConfig.Spec.Customization.Brand
+}
+
+// Get the brand from the console-config in the data of the console CM
+func getConsoleBrand(t *testing.T, client *testframework.Clientset) operatorsv1.Brand {
+	cm, err := testframework.GetConsoleConfigMap(client)
+	if err != nil {
+		t.Fatalf("error: %s", err)
+	}
+
+	data := cm.Data["console-config.yaml"]
+	brandingValue := ""
+	temp := strings.Split(data, "\n")
+	for _, item := range temp {
+		if strings.Contains(item, "branding") {
+			brandingValue = strings.Split(strings.TrimSpace(item), ":")[1]
+		}
+	}
+	brand, _ := stringToBrand(strings.TrimSpace(brandingValue))
+	return brand
+}
+
+// Helper function to convert string version of valid brands to their corresponding constant name
+func stringToBrand(brandstr string) (b operatorsv1.Brand, ok bool) {
+	switch found := operatorsv1.Brand(brandstr); found {
+	case
+		operatorsv1.BrandDedicated,
+		operatorsv1.BrandOnline,
+		operatorsv1.BrandOKD,
+		operatorsv1.BrandOCP,
+		operatorsv1.BrandOpenShift,
+		operatorsv1.BrandAzure:
+		return found, true
+	default:
+		fmt.Printf("invalid brand (%s), defaulting to %s \n", brandstr, operatorsv1.BrandOKD)
+		return operatorsv1.BrandOKD, false
+	}
+}
+
+// Helper function that decides whether to update a config map (if it exists) or create a new one
+func modifyConsoleConfigMap(client *testframework.Clientset, ns string, cm *v1.ConfigMap) (*v1.ConfigMap, error) {
```

Comment:
`modify` implies it would change an existing, not create a new.  Best name?

Reply:
Comment seems to be needed only because the func name is unclear / implies something different.

Reply:
What about `updateOrCreateConsoleConfigMap`?